internal/notifier/grafana.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/fluxcd/pkg/runtime/events"
+	"github.com/hashicorp/go-retryablehttp"
+
+)
+
+// Discord holds the hook URL
+type Grafana struct {
+	URL      string
+	Token    string
+	ProxyURL string
+}
+
+// GraphiteAnnotation represents a Grafana API annotation in Graphite format
+type GraphitePayload struct {
```

Comment:
I'm familiar with Grafana's API, but does this mean people must use Graphite for Flux metrics? Isn't Prometheus supported?

Reply:
Graphana API receives accepts to different payloads, the graphana [one](https://grafana.com/docs/grafana/latest/http_api/annotations/#create-annotation) and the graphite [one](https://grafana.com/docs/grafana/latest/http_api/annotations/#create-annotation-in-graphite-format). The payloads are similar but the graphite one has fewer fields. 
Prometheus is not supported by this API, but I can work on a different provider for Prometheus to be used by Grafana annotations.  
This provider stores the annotations in the Grafana DB (most of the times ephemeral).

Reply:
We need to create a step-by-step guide for people to follow, with print screens from Grafana. Maybe a section in here https://fluxcd.io/docs/guides/monitoring/

Reply:
I will start working on this asap

Reply:
https://github.com/fluxcd/website/pull/747 @stefanprodan I don't know if this was what you had in mind